libr/core/cmd.c

```diff
@@ -6836,10 +6836,30 @@ R_API int r_core_cmd_lines(RCore *core, const char *lines) {
 	if (!odata) {
 		return false;
 	}
+	int line_count = 0;
+	{
+		int i = 0;
+		while (lines[i]) {
+			if (lines[i] == '\n') {
+				line_count++;
+			}
+			i++;
+		}
+	}
+
+#if __UNIX__
+	const bool show_progress_bar = core->print->enable_progressbar && r_config_get_i (core->config, "scr.interactive") && r_config_get_i (core->config, "scr.progressbar") && r_cons_isatty ();
```

Comment:
reduce the amount of duplicated code in here. something like this will look better:

```c
#if __UNIX__
 	 const bool istty = r_cons_isatty ();
#else
         const bool istty = true
 #endif

const bool show_progress_bar = core->print->enable_progressbar && r_config_get_i (core->config, "scr.interactive") && r_config_get_i (core->config, "scr.progressbar") && istty);
```

also move all this logic into a separate function so the condition can be checked in other places and the code will be more readable. at the end no #if code must be in here.

```
const show_progress_bar = want_progress_bar(core);
```